### PR TITLE
Compute ROI values from voxelised shape

### PR DESCRIPTION
**Pull request by robbietuk**
Addresses #914 

Overloads `compute_total_ROI_values` and `compute_ROI_values_per_plane`The new functions take `discretised_shape` (a `VoxelsOnCartesianGrid<float>` object) as argument. This was implemented as 75% of ROI value computation is creating the shapes. This change allows that command to be run once.

### CODE REVIEW

**Comment by robbietuk** [comment]
This is a seperate PR to #913 but we should wait until that PR is merged and then expand the `ROI_demo.py` to include a demonstration of this functionality.

File: src/include/stir/evaluation/compute_ROI_values.h
Comment (KrisThielemans) [inline src/include/stir/evaluation/compute_ROI_values.h:69]:
```suggestion
                         const VoxelsOnCartesianGrid<float>& discretised_shape,
```

File: src/eval_buildblock/compute_ROI_values.cxx
Comment (KrisThielemans) [inline src/eval_buildblock/compute_ROI_values.cxx:143]:
```suggestion
                         const VoxelsOnCartesianGrid<float>& discretised_shape,
```

File: src/eval_buildblock/compute_ROI_values.cxx
Comment (KrisThielemans) [inline src/eval_buildblock/compute_ROI_values.cxx:147]:
```suggestion
    compute_ROI_values_per_plane(values, image, discretised_shape, num_samples);
```

File: documentation/release_5.0.htm
Comment (KrisThielemans) [inline documentation/release_5.0.htm:255]:
```suggestion
    Overloaded <code>compute_total_ROI_values</code> and <code>compute_ROI_values_per_plane</code> functions to allow the passing of  a<code>VoxelsOnCartesianGrid</code> object instead of a <code>Shape3D</code> as the ROI shape.
```

**Review by KrisThielemans — CHANGES_REQUESTED** [review]
some comments, but I think the main one is that the new versions actually don't use `num_samples`. In fact, they require that the geometric info between image and `discretised_shape` are identical (something which is taken care of in the original version). So the new versions need to do a check on that. (There's a method for that I believe).

Also, to avoid a lot of casting in client code, I think it is better to have the `discretised_shape` arguments as `DiscretisedDensity<3,float>&` (and do the cast inside if it's needed).

**Comment by robbietuk** [comment]
@KrisThielemans I thought drafts dont run travis... could you please kill these jobs as I forgot to use [ci skip]

**Comment by robbietuk** [comment]
> (and do the cast inside if it's needed).

`test_ROIs` works locally and my python code works without the dynamic cast in the 
```
compute_ROI_values_per_plane(VectorWithOffset<ROIValues>& values,
                             const DiscretisedDensity<3,float>& density,
                             const DiscretisedDensity<3,float>& discretised_shape)
```
function. The dynamic cast to a `VoxelsOnCartesianGrid` is needed for the generation of the shape (due to voxel sizes?) and then all comparisons in the function are performed between elements of the arrays. Please correct me if I am wrong, my understanding of volume objects in STIR is limited.

Awaiting travis.

**Pull request merged**
